cli/tests/activate.bats

```diff
@@ -132,6 +132,10 @@ setup() {
   export _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/empty.json"
 }
 teardown() {
+  # Cleaning up the `BATS_TEST_TMPDIR` occasionally fails,
+  # because of an 'env-registry.json' that gets concurrently written
+  # by the watchdog as the activation terminates.
+  wait_for_watchdogs
```

Comment:
Do we need this in every file that runs `flox activate`?

Reply:
the "issue" with the tests in `activate.bats` came to be because we used `setup_isolated_flox`, which causes the env-registry in particular to be written in `BATS_FILE_TMPDIR`. In tests without `setup_isolated_flox` the files will be cleaned up all together at the suite teardown giving the last watchdog more time to flush the registry file.

Reply:
Note from pairing: there's a good chance that we introduce some combination of `setup_isolated_flox` and `flox activate` in other test files in the future but we can't see any way to guard against that by code or convention. Even leaving a comment in `setup_isolated_flox` is highly likely to be overlooked.